internal/imex/imex.go

```diff
@@ -0,0 +1,81 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package imex
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	spec "github.com/NVIDIA/k8s-device-plugin/api/config/v1"
+)
+
+// Channels represents a set of IMEX channels.
+type Channels []*Channel
+
+// Channel represents an IMEX channel.
+type Channel struct {
+	ID       string
+	Path     string
+	HostPath string
+}
+
+// GetChannels returns the set of channels for the given config.
+// If the selection of the default IMEX channel is disabled no channels are returned.
+func GetChannels(config *spec.Config, devRoot string) (Channels, error) {
+	var channels Channels
+	for _, channelID := range config.Imex.ChannelIDs {
+		id := fmt.Sprintf("%d", channelID)
+		channelName := "channel" + id
+		path := filepath.Join("/dev/nvidia-caps-imex-channels", channelName)
+		channel := Channel{
+			ID:       id,
+			Path:     path,
+			HostPath: filepath.Join(devRoot, path),
+		}
+		if exists, err := channel.exists(); !exists {
+			if config.Imex.Required {
+				return nil, errors.Join(err, fmt.Errorf("requested IMEX channel %v does not exist", channelName))
+			}
+			klog.Warningf("Ignoring requested IMEX channel %v (%v)", channelName, err)
+			continue
+		}
+		klog.Infof("Selecting IMEX channel %v", channelName)
+		channels = append(channels, &channel)
+	}
+	return channels, nil
+}
+
+// exists checks whether the channel exists at the host path.
+func (c Channel) exists() (bool, error) {
```

Comment:
```suggestion
func (c Channel) hostPathExists() (bool, error) {
```

Reply:
I have updated the implementation and added a more extensive docstring.

Reply:
Looks good.